src/stream/src/executor/group_top_n.rs

```diff
@@ -130,11 +130,8 @@ pub struct InnerGroupTopNExecutorNew<S: StateStore, const WITH_TIES: bool> {
     /// group key -> cache for this group
     caches: HashMap<Vec<Datum>, TopNCache<WITH_TIES>>,
 
-    #[expect(dead_code)]
-    /// Indices of the columns on which key distribution depends.
-    key_indices: Vec<usize>,
-
-    sort_key_len: usize,
+    /// The length of the ORDER BY clause. Only used when `WITH_TIES` is true.
+    order_by_len: usize,
```

Comment:
Could it be better to separate the _actual_ order-by columns and the PK columns? I had considered that before this PR

Reply:
https://github.com/risingwavelabs/risingwave/pull/4682 introduced `catalog.Table` in the proto. What's your idea? cc @shmiwy